Sources/TestingMacros/CMakeLists.txt

```diff
@@ -0,0 +1,72 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2024 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+cmake_minimum_required(VERSION 3.19.6)
```

Comment:
Hmm, shipping the macro is going to be interesting since it's sort of part of the library. If we are working in a cross-compiling situation where we're on machine A, building the tests to run on machine B, but we're building tooling that runs on machine C, then the macro will need to be cross-compiled for machine C, while the library is compiled for machine B. Since this is a separate project, I think it will work from a third-party's project, though they'll still need to build the macros for their environment... Something like this (in their code, not for you) should probably work:

```cmake
ExternalProject_Add(SwiftTestingMacros
  GIT_REPOSITORY https://github.com/apple/swift-testing
  SOURCE_SUBDIR Sources/TestingMacros
  ...)
```

Just something to think about since it means that we're not necessarily running the macro build from
```
ExternalProject_Add(TestingMacros
   CMAKE_CACHE_DEFAULT_ARGS "-DCMAKE_MODULE_PATH:FILEPATH=${CMAKE_MODULE_PATH}"
   PREFIX "tm"
   SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/TestingMacros"
   INSTALL_COMMAND "")
 ExternalProject_Get_Property(TestingMacros BINARY_DIR)
```

which then means that we may or may not have the cmake module path coming from the wrapping project. Also, given the cross-compilation story, I'm not certain the `Settings` will always be the same here.

Reply:
Yes, this is modeled on [`this example`](https://github.com/apple/swift-cmake-examples/blob/7a86d39fe88271501431f5f7214edbcc5c1f745b/4_swift_macros/CMakeLists.txt#L26-L34).  

I agree passing the module path here was a bit of a hack.  Do you think I should just duplicate the settings between the main and macros project?

Reply:
Yeah, so the difference here and that example is that in the example, I really just intended to use the macro within the project. With Swift-Testing, we're vending the macros too, not as an implementation detail, so we need some way for the project consuming Swift-Testing to also build a set of macros on their system. They shouldn't have to rebuild all of Swift-Testing from scratch if we've already provided them the pre-compiled library, just so that they can have the macros.

I'd like to make the ability to make the macros buildable independently of the rest of Swift-Testing relatively painless. Perhaps the shared bits can go one layer deeper?
If we move the shared bits in `Swift-Testing` to `cmake/module/shared/AvailabilityDefinitions.cmake`, then from `Sources/TestingMacros/CMakeLists.txt`, we can `list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules ${CMAKE_CURRENT_SOURCE_DIR}/../cmake/modules/shared)`, and the library content can have `list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/shared)`. That way we can control what modules are shared between both the macro and library parts of the build, and what we want to keep isolated. That way the `ExternalProject_Add` isn't responsible for forwarding that location into the build, the macro build knows where to look for its modules and will configure and build in isolation too. @compnerd, tell me if I've lost my mind here.

Reply:
I've made an attempt.  Let me know what you think.  Again, happy to iterate.